ci/scripts/towncrier_automation.py

```diff
@@ -0,0 +1,74 @@
+#!python3
+from __future__ import annotations
+
+import argparse
+import subprocess
+
+from packaging.version import parse
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        prog="towncrier-automation",
+        description="This script runs townncrier for a given version, creates a branch off of the current one, and then creates a PR into the original branch with the changes.  The PR will be backported to main if the current branch is not main.",
+        usage="python towncrier_automation.py --version <version> [--dry-run]",
+    )
+    parser.add_argument("--version", type=str, help="What the new version is")
+    parser.add_argument(
+        "--dry-run",
+        help="Whether or not to dry-run the actual creation of the pull request",
+        action="store_true",
+    )
+    args = parser.parse_args()
+    version = args.version
+    parse(args.version)
+
+    # Run towncrier
+    if subprocess.call(["towncrier", "build", f"--version={version}", "--yes"]):
```

Comment:
isn’t this inverse? if the command succeeds, the status code is `0`, but this branch is entered

Reply:
Ah this must have gotten lost in the sauce...seems like I actually lost one or two things.  I had `retcode` here

Reply:
Ah I must have committed to the wrong branch.  I also got rid of `call` right